src/stan/lang/ast/node/algebra_solver.hpp

```diff
@@ -0,0 +1,64 @@
+#ifndef STAN_LANG_AST_NODE_ALGEBRA_SOLVER_HPP
+#define STAN_LANG_AST_NODE_ALGEBRA_SOLVER_HPP
+
+#include <stan/lang/ast/node/expression.hpp>
+#include <string>
+
+namespace stan {
+  namespace lang {
+
+    struct expression;
+
+    /**
+     * Structure for algebraic solver statement.
+     */
+    struct algebra_solver {
+      /**
+       * Name of the algebra system.
+       */
+      std::string system_function_name_;
+
+      /**
+       * Initial guess for solution.
+       */
+      expression x_;
+
+      /**
+       * Parameters.
+       */
+      expression y_;
+
+      /**
+       * Real-valued data.
+       */
+      expression dat_;
+
+      /**
+       * Integer-valued data.
+       */
+      expression dat_int_;
+
+      /**
+       * Construct a default algebra solver node.
+       */
+      algebra_solver();
+
+      /**
+       * Construct an algebraic solver.
+       *
+       * @param system_function_name name of ODE system
+       * @param x initial guess for solution
```

Comment:
sorry about that, will adopt the convention used in the manual:
y - initial guess / solution
theta - parameters
x_r and x_i for data